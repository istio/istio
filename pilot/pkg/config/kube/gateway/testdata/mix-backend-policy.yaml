apiVersion: gateway.networking.k8s.io/v1alpha3
kind: BackendTLSPolicy
metadata:
  name: tls-upstream-echo
  namespace: default
spec:
  targetRefs:
    - kind: Service
      name: echo
      group: ""
  validation:
    caCertificateRefs:
      - kind: ConfigMap
        name: auth-cert
        group: ""
    hostname: auth.example.com
---
# A redundant policy for the same service
apiVersion: gateway.networking.k8s.io/v1alpha3
kind: BackendTLSPolicy
metadata:
  name: tls-upstream-echo-extra
  namespace: default
spec:
  targetRefs:
    - kind: Service
      name: echo
      group: ""
  validation:
    subjectAltNames:
      - type: Hostname
        hostname: "extra.com"
    caCertificateRefs:
      - kind: ConfigMap
        name: auth-cert
        group: ""
    hostname: auth-extra.example.com
---
apiVersion: gateway.networking.x-k8s.io/v1alpha1
kind: XBackendTrafficPolicy
metadata:
  name: lb-policy
  namespace: default
spec:
  targetRefs:
    - kind: Service
      name: echo
      group: ""
  sessionPersistence:
    sessionName: foo
    absoluteTimeout: 1h
    type: Cookie
    cookieConfig:
      lifetimeType: Permanent

