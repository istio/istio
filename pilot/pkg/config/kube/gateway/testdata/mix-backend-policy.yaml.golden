apiVersion: networking.istio.io/v1
kind: DestinationRule
metadata:
  annotations:
    internal.istio.io/parents: XBackendTrafficPolicy/default.lb-policy,BackendTLSPolicy/default.tls-upstream-echo
  creationTimestamp: null
  name: echo-istio-autogenerated-k8s-gateway
  namespace: default
spec:
  host: echo.default.svc.domain.suffix
  trafficPolicy:
    loadBalancer: {}
    tls:
      credentialName: configmap://default/auth-cert
      mode: SIMPLE
      sni: auth.example.com
---
